documentation/source/CONTRIBUTING.md

```diff
@@ -1,88 +1,101 @@
-
-    * To run pre-commit on all files just run
-   ```
-   $ pre-commit run --all-files
-   ```
--->
-7. Create a pull request against <b>master</b> branch.
 
 
+## Code Style
 
-## Jupyter Notebooks Contribution
+We follow the **reStructuredText** docstring format (default of PyCharm), along with typing.
+```python
+def python_function(first_argument: int, second_argument: int) -> str:
+    """Do something with the two arguments.
 
-Pulling updates from remote might cause merge conflicts with jupyter notebooks. The tool [nbdime](https://nbdime.readthedocs.io/en/latest/) might solve this.
-* Installing nbdime
-```
-pip install ndime
-```
-* Run a diff between two notebooks
-```
-nbdiff notebook_1.ipynb notebook_2.ipynb
+    :param first_argument: First argument to the function
+    :param second_argument: Second argument to the function
+    :return: Description of the output
+    """
 ```
 
-## Code Style Guidelines
 
-We are working hard to make sure all the code in this repository is readable, maintainable and testable.
-We follow the Google docstring guidelines outlined on this [styleguide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings) page. For example:
-```python
-  def python_function(first_argument: int, second_argument: int) -> bool:
-      """
-      This function will do something with the two arguments.
 
-      Args:
-          first_argument (int): the first argument to the function
-          second_argument (int): the second argument to the function
 
-      Raises:
-          Exception: arguments should be ints not floats
+## Code Formatting
+
+We enforce [black](https://github.com/psf/black) code formatting in addition to existing flake8 checks. 
+
+To ensure everyone uses same code style, a project-wise [configuration file](https://github.com/Deci-AI/super-gradients/blob/master/pyproject.toml) has been added to SG repo. It ensures all formatting will be exactly the same regardless of OS, python version or the place where code formatting check is happening. 
+
+### Installation
+
+To start, one need to install required development dependencies (actual versions of black, flake8 and git commit hooks):
+
+`$ pip install -r requirements.dev.txt`
 
-      Returns:
-          bool: whether or not the calculation was correct
-      """
+### Pre-Commit Hooks
+
+A pre-commit hook as an easy way to ensure all files in the commit are already formatted accordingly and pass linter checks. If they are not, the git will prevent commit of problematic files unless errors are fixed.
+
+To start, run the following command from SG repo root:
+```bash
+$ pip install pre-commit
+$ pre-commit install
 ```
 
+The command should complete without errors. Once done, all your upcoming commits will be checked via black & flake8. 
+
+### Usage
+
+Just run ```$ black .``` from the SG root. It will reformat the whole repo.
```

Comment:
Add our flake8 command from the CI.

Reply:
Added command in b3f061ac6b3420c2012ccb3af7c7cf8a86a0da82